File: playwright/pages/AdminPage.ts
import { expect, Locator, Page } from "@playwright/test";
import { BasePage } from "./BasePage";

export class AdminPage extends BasePage {
  readonly customerTable: Locator;
  readonly auditTable: Locator;

  constructor(page: Page) {
    super(page);
    this.customerTable = this.testId("customer-table");
    this.auditTable = this.testId("audit-table");
  }

  async openDashboard(): Promise<void> {
    await this.openView("admin-dashboard", /Operations/i);
  }

  async openCustomers(): Promise<void> {
    await this.openView("admin-customers", /Customers/i);
    await expect(this.customerTable).toBeVisible();
  }

  async openAudit(): Promise<void> {
    await this.openView("admin-audit", /Audit/i);
    await expect(this.auditTable).toBeVisible();
  }

  async expectMetricsVisible(): Promise<void> {
    await expect(this.page.getByText(/Customers/i).first()).toBeVisible();
  }
}

File: selenium/src/test/java/com/bankingsystem/qa/pages/AdminPage.java
package com.bankingsystem.qa.pages;

import org.openqa.selenium.WebDriver;

public class AdminPage extends BasePage {

    public AdminPage(WebDriver driver) {
        super(driver);
    }

    public AdminPage openDashboard() {
        openView("admin-dashboard");
        return this;
    }

    public AdminPage openCustomers() {
        openView("admin-customers");
        find("customer-table");
        return this;
    }

    public AdminPage openAudit() {
        openView("admin-audit");
        find("audit-table");
        return this;
    }

    public String customerTableText() {
        return find("customer-table").getText();
    }

    public String auditTableText() {
        return find("audit-table").getText();
    }

    /** Open user administration without expecting it to render data. */
    public AdminPage openUsersExpectingDenial() {
        openView("admin-users");
        return this;
    }

    public String viewText() {
        return find("view").getText();
    }
}
